src/daemon/daemon.cpp

```diff
@@ -2053,33 +2052,36 @@ void mp::Daemon::install_sshfs(VirtualMachine* vm, const std::string& name)
 
     SSHSession session{vm->ssh_hostname(), vm->ssh_port(), vm->ssh_username(), key_provider};
 
-    mpl::log(mpl::Level::info, category, fmt::format("Installing sshfs in \'{}\'", name));
+    mpl::log(mpl::Level::info, category, fmt::format("Installing the multipass-sshfs Snap for \'{}\'", name));
 
-    int retries{0};
-    while (++retries <= max_install_sshfs_retries)
+    // Check if Snap support is installed in the instance
+    auto proc = session.exec("which snap");
+    if (proc.exit_code() != 0)
     {
-        try
-        {
-            auto proc = session.exec("sudo apt update && sudo apt install -y sshfs");
-            if (proc.exit_code(std::chrono::minutes(5)) != 0)
-            {
-                auto error_msg = proc.read_std_error();
-                mpl::log(mpl::Level::warning, category,
-                         fmt::format("Failed to install 'sshfs', error message: '{}'", mp::utils::trim_end(error_msg)));
-            }
-            else
-            {
-                break;
-            }
-        }
-        catch (const mp::ExitlessSSHProcessException&)
+        mpl::log(mpl::Level::warning, category, fmt::format("Snap support is not installed for \'{}\'", name));
+        throw std::runtime_error(
+            fmt::format("Snap support needs to be installed for \'{}\' in order to support mounts. "
+                        "Please see https://docs.snapcraft.io/installing-snapd for information on "
+                        "how to install Snap support for your instance's distribution.",
+                        name));
+    }
+
+    try
+    {
+        auto proc = session.exec("sudo snap install multipass-sshfs");
+        if (proc.exit_code(std::chrono::minutes(5)) != 0)
         {
-            mpl::log(mpl::Level::info, category, fmt::format("Timeout while installing 'sshfs' in '{}'", name));
+            auto error_msg = proc.read_std_error();
+            mpl::log(mpl::Level::warning, category,
+                     fmt::format("Failed to install \'multipass-sshfs\', error message: \'{}\'",
+                                 mp::utils::trim_end(error_msg)));
```

Comment:
This doesn't go into the error the user sees, wonder if it should? OTOH `-v` makes it show…

Reply:
Hmm, I suppose it should be surfaced to the client to give a clue to the user why it didn't install.

Reply:
I've left this alone for now.  If you think it's necessary to surface to the client without `-v`, then I'll look into it.